### PR TITLE
Support `#allow-multiple!` predicate to enable multiple content ranges

### LINKED ISSUE #1481: Support `#allow-multiple!` predicate to enable multiple content ranges
- [ ] Modify `getTarget: Target` field https://github.com/cursorless-dev/cursorless/blob/417e4f9637d8204e671fa34af788dca87cdd5686/packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/scope.types.ts#L38 to `getTargets(): Target[]`
- [ ] Add predicate `#allow-multiple!` that sets a `allowMultiple` field on the node output 
- [ ] When processing scopes in base tree sitter scope handler https://github.com/cursorless-dev/cursorless/blob/417e4f9637d8204e671fa34af788dca87cdd5686/packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/TreeSitterScopeHandler/BaseTreeSitterScopeHandler.ts#L43 detect if there are multiple scopes with identical domains.
- [ ] If it happens, check the nodes for an `allowMultiple` field.  If it exists, merge into one scope that has multiple content ranges.  If it doesn't exist, throw an error saying they need to use the predicate if it was intentional

### PR DESCRIPTION
**Pull request by pokey**
- Depends on https://github.com/cursorless-dev/cursorless/pull/1509
- Fixes #1481

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/TreeSitterScopeHandler/TreeSitterScopeHandler.ts [hunk 1]
```diff
@@ -33,38 +35,49 @@ export class TreeSitterScopeHandler extends BaseTreeSitterScopeHandler {
   protected matchToScope(
```
Comment (pokey) [inline packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/TreeSitterScopeHandler/TreeSitterScopeHandler.ts:35]: this function turned out pretty ugly.  I wonder if I should have another version of `getRelatedRange` that just returns a `range` like it used to, as don't care about `allowMultiple` for any of the related ranges other than `.iterationScope`

File: packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts [hunk 2]
```diff
@@ -99,6 +100,9 @@ export class TreeSitterQuery {
           range: captures
             .map(({ range }) => range)
             .reduce((accumulator, range) => range.union(accumulator)),
+          allowMultiple: captures
```
Comment (AndreasArvidsson) [inline packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts:103]: Why not `captures.some(capture => capture.allowMultiple)`?

**Review by AndreasArvidsson — APPROVED** [review]

File: packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts (hunk 2 above)
Comment (pokey) [reply packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts:103]: Oh yeah that's the way to go. Not sure how I landed here 🤔. I blame copilot 😅

Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts:103]: Sure blame the servants :p

**Pull request merged**